cupyx/jit/cg.py

```diff
@@ -284,5 +308,139 @@ def call_const(self, env):
         return _Data(f'cg::this_{self.group_type}()', cg_type)
 
 
+class _Sync(_BuiltinFunc):
+
+    def __call__(self, group):
+        """Calls ``cg::sync()``.
+
+        Args:
+            group: a valid cooperative group
+
+        .. seealso:: `cg::sync`_
+
+        .. _cg::sync:
+            https://docs.nvidia.com/cuda/archive/11.6.0/cuda-c-programming-guide/index.html#collectives-cg-sync
+        """
+        super().__call__()
+
+    def call(self, env, group):
```

Comment:
How about checking `isinstance(group.ctype, _ThreadGroup)` here?

Reply:
Done in 6e759600.